### ISSUE TITLE
Suggestions hang after sklearn's PCA fit_transform call

### ISSUE THREAD

**Issue by chmielcode** [microsoft/pylance-release#2296]
## Environment data

-   Language Server version: 2022.1.3 (pyright 98868f26)
-   OS and version: Windows 10
-   Python version (& distribution if applicable, e.g. Anaconda): 3.9.9

## Actual behaviour
Suggestions are "loading" indefinitely after the fit_transform() call (valid one with arguments or as shown below in the code snippet) and the CPU usage is high.
Couple of seconds after commenting out ".fit_transform()" everything works again.

## Code Snippet / Additional information
```python
import pandas as pd
from sklearn import decomposition
decomposition.PCA(n_components=3).fit_transform()

# type e.g. "pd.Da" and press ctrl+space
```

**Comment by erictraut** [comment]
Thanks for the bug report. I've root caused the problem and have a potential solution. It affects the return type inference logic, which is used extensively in an untyped library like sklearn. The proposed change has some (small) regression risk associated with it, so need to discuss the best way to validate and release it.

**Comment by chmielcode** [comment]
Thank you. After updating to the just released 2022.1.5 (pyright 8d096292) I cannot trigger this bug anymore.
